src/network/network.cpp

```diff
@@ -486,6 +486,33 @@ void ParseConnectionString(const char **company, const char **port, char *connec
 	}
 }
 
+/**
+ * Finds the first parameter and value in a query string
+ *  Format: param0=val0&param1=val1&...
+ *
+ * query will be moved to the start of the next query, and parameter name and
+ * value will be set to the name and value strings given by the user, inside
+ * the memory area originally occupied by connection_string.
+ */
+bool ParseQueryString(const char **name, const char **value, char **query)
```

Comment:
The doc block should have `@param` and `@return` lines explaining the parameters. When I see just the signature I think of a function that fills an array of pointers, and then get worried that the length of those arrays is not specified anywhere.
After reading closely I see that it only reads a single element at a time.
I'd suggest changing the signature to take references to pointers instead, to make it more clear the `name` and `value` parameters are "out" (also document that), and `query` is in/out.

Reply:
TBH, I prefer the "pointers to variable" method of out parameter - it's much more obvious that it's an out parameter when you're calling the function. Also has the potential to make them optional, assuming you remember to check for null :)

Reply:
The other way would be much more C++'y: Make an iteration structure, with the iterator dereferencing to a pair of std::strings.

Reply:
I actually started writing this using std::strings because I'm more used to that but converting them to char* and making sure they weren't going out of scope was tricky so I just left used char* the whole way.